src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -308,6 +315,13 @@ public static PackageSpec GetPackageSpec(IEnumerable<IMSBuildItem> items)
                 result.RestoreMetadata.CentralPackageVersionOverrideDisabled = isCentralPackageVersionOverrideDisabled;
                 result.RestoreMetadata.CentralPackageFloatingVersionsEnabled = isCentralPackageFloatingVersionsEnabled;
                 result.RestoreMetadata.CentralPackageTransitivePinningEnabled = isCentralPackageTransitivePinningEnabled;
+
+                string isSdk = specItem.GetProperty("UsingMicrosoftNETSdk");
+
+                if (bool.TryParse(isSdk, out bool value))
+                {
+                    result.UsingMicrosoftNETSdk = value;
+                }
```

Comment:
Since you took up my previous suggestion of making PackageSpec.cs default to true, that means this here now has a bug, because when the project file doesn't contain the property, the value is keeping its default `true` value, rather than being set to `false`.

```suggestion
                bool usingMicrosoftNETSdk = false;
                if (bool.TryParse(isSdk, out bool value))
                {
                    usingMicrosoftNETSdk = value;
                }
                result.UsingMicrosoftNETSdk = usingMicrosoftNETSdk;
```